config.json

```diff
@@ -1,24 +1,69 @@
 {
     "port": 8000,
     "listenOn": [],
-    "regions": {
-        "ap-northeast-1": ["s3.ap-northeast-1.amazonaws.com"],
-        "ap-southeast-1": ["s3.ap-southeast-1.amazonaws.com"],
-        "ap-southeast-2": ["s3.ap-southeast-2.amazonaws.com"],
-        "eu-central-1": ["s3.eu-central-1.amazonaws.com",
-                         "s3.eu.central-1.amazonaws.com"],
-        "eu-west-1": ["s3.eu-west-1.amazonaws.com"],
-        "sa-east-1": ["s3.sa-east-1.amazonaws.com"],
-        "us-east-1": ["s3.amazonaws.com",
-                      "s3-external-1.amazonaws.com",
-                      "s3.us-east-1.amazonaws.com"],
-        "us-west-1": ["s3.us-west-1.amazonaws.com"],
-        "us-west-2": ["s3-us-west-2.amazonaws.com"],
-        "us-gov-west-1": ["s3-us-gov-west-1.amazonaws.com",
-                          "s3-fips-us-gov-west-1.amazonaws.com"],
-        "localregion": ["localhost"],
-        "test-region": ["s3.scality.test"],
-        "docker-region": ["s3.docker.test"]
+    "locationConstraints": {
+        "aws-us-east-1": {
+            "type": "aws_s3",
+            "legacyAwsBehavior": true,
+            "details": {
+                "region": "us-east-1",
```

Comment:
1) it is possible to have legacyAwsBehavior true and us-east-1.  we are controlling the behavior (errors, etc.).  the data is just being stored on AWS in the background.

2) in fact, we could get rid of region altogether in the aws_s3 type.  if we have the endpoint that is enough.  we just need to know where to call.

3) let's keep legacyAwsBehavior at the current level (same as type and details).  the details are really about the details to hit the right data endpoint.  legacyAwsBehavior is about behavior.